spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/AbstractRepositoryRestResourceIT.java

```diff
@@ -0,0 +1,168 @@
+package io.quarkus.ts.spring.data.rest;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.http.HttpStatus;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import io.quarkus.ts.spring.data.AbstractDbIT;
+import io.restassured.response.ValidatableResponse;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public abstract class AbstractRepositoryRestResourceIT extends AbstractDbIT {
+
+    protected abstract String getUrl();
+
+    protected abstract List<String> getOriginalItems();
+
+    protected abstract String getCreatedItem();
+
+    protected abstract long getCreatedItemId();
+
+    protected abstract String getUpdatedItem();
+
+    protected abstract String getItemNameHalJsonPath();
+
+    protected abstract String getItemIdUrl(long id);
+
+    @Order(1)
+    @Test
+    void getAll() {
+        //GET - List all items
+        ValidatableResponse response = app.given()
+                .accept("application/json")
+                .when().get(getUrl())
+                .then()
+                .statusCode(HttpStatus.SC_OK);
+        getOriginalItems().stream()
+                .map(CoreMatchers::containsString)
+                .forEach(response::body);
+    }
+
+    @Order(2)
+    @Test
+    void getAllHal() {
+        //GET - List all items in hal+json format
+        String articlesList = app.given()
+                .accept("application/hal+json")
+                .when().get(getUrl())
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract().response().jsonPath().getString(getItemNameHalJsonPath());
```

Comment:
Wouldn't it be easier to get a list and to compare it with another list below?

Reply:
Good point, fixed.